dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -34,33 +36,119 @@
 import org.apache.bcel.classfile.InnerClass;
 import org.apache.bcel.classfile.InnerClasses;
 import org.apache.bcel.classfile.JavaClass;
+import org.apache.bcel.classfile.Method;
 import org.apache.bcel.util.ClassPath;
 import org.apache.bcel.util.SyntheticRepository;
 
 /**
- * This class reads a jar file and lists method references in its class files.
+ * This class reads jar files and runs static linkage analysis on them
  */
-class JarDumper {
+class StaticLinkageChecker {
+
+  /**
+   * Given list of the jar files as file names in filesystem, outputs the report of static linkage
+   * check.
+   *
+   * @param arguments jar files available in the file system
+   * @throws IOException when there is a problem in reading a jar file
+   * @throws ClassNotFoundException when there is a problem in reading a class from a jar file
+   */
+  public static void main(String[] arguments) throws IOException, ClassNotFoundException {
+    String report = generateStaticLinkageReport(arguments);
+    System.out.println(report);
+  }
+
+  @VisibleForTesting
+  static String generateStaticLinkageReport(String[] jarFileNames) throws IOException,
+      ClassNotFoundException {
+    StringBuilder stringBuilder = new StringBuilder();
+    Set<FullyQualifiedMethodSignature> externalMethodReferences = new HashSet<>();
+    List<Path> paths = new ArrayList<>();
+    for (String jarFileName : jarFileNames) {
+      File jarFile = new File(jarFileName);
```

Comment:
If these are file names, what directory are these files in? A file name alone is not enough to locate a Jar.

Reply:
Now it uses Path class, and javadoc mentions they're absolute paths.